src/manifest.c

```diff
@@ -239,9 +240,20 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		}
 		if (strncmp(line, "filecount:", 10) == 0) {
 			filecount = strtoull(c, NULL, 10);
+			if (filecount > 1000000) {
+				fprintf(stderr, "Error: preposterous (%" PRIu64 ") number of files in %s Manifest, more than a million skipping\n",
+					filecount, component);
+				goto err_close;
+			}
 		}
 		if (strncmp(line, "contentsize:", 12) == 0) {
 			contentsize = strtoull(c, NULL, 10);
+			/* if (contentsize > 2000000000) {
```

Comment:
Is there a reason this has been commented out? I think 2TB is a pretty reasonable number right now considering 21300 full contentsize is just over 14GB.

Reply:
Yes, it makes too many tests fail!
The content-size in the MoM is pretty bogus - 13,805,671,819

Reply:
That... is wrong. If you look at published MoMs (https://download.clearlinux.org/update/21410/Manifest.MoM for one) contentsize is always created as 0.

Reply:
I'm now seeing all of our test cases define MoM contentsize to that number... those should all be 0, I wonder why they were initialized to that at all.